modules/core/src/main/java/org/locationtech/jts/geom/CoordinateSequence.java

```diff
@@ -111,15 +145,48 @@
    * @return the value of the Y ordinate in the index'th coordinate
    */
   double getY(int index);
+  
+  /**
+   * Returns ordinate Z of the specified coordinate if available.
+   * 
+   * @param index
+   * @return the value of the Z ordinate in the index'th coordinate, or Double.NaN if not defined.
+   */
+  default double getZ(int index) {
+     if( (getDimension()-getMeasures()) > 2 ) {
+	 return getOrdinate( index, 2 );
+     }
+     else {
+         return Double.NaN;
+     }
+  }
 
+  /**
+   * Returns ordinate M of the specified coordinate if available.
+   * 
+   * @param index
+   * @return the value of the Z ordinate in the index'th coordinate, or Double.NaN if not defined.
+   */
+  default double getM(int index) {
+     if( getDimension() > 2 && getMeasures() > 0 ) {
+	 final int mIndex = getDimension()-getDimension();
```

Comment:
should this be `getDimension() - getMeasures()`?

Reply:
yes, yes it should. And I should have test cases before this goes live :)